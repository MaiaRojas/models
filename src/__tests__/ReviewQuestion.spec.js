const mongoose = require('mongoose');
const { ReviewQuestion } = require('../../')(mongoose);

describe('ReviewerQuestion', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await ReviewQuestion.createIndexes();
  });

  it('should fail with missing props', () => {
    const doc = new ReviewQuestion();
    return doc.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail when type is unknown', () => {
    const doc = new ReviewQuestion({
      i18nId: 'dropout',
      type: 'other',
      visibility: 'public',
      options: ['foo', 'bar'],
    });

    return doc.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail when visibility is unknown', () => {
    const reviewQuestion = new ReviewQuestion({
      i18nId: 'dropout',
      type: 'open',
      visibility: 'other',
    });

    return reviewQuestion.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const reviewQuestion = new ReviewQuestion({
      i18nId: 'dropout',
      type: 'multipleChoice',
      visibility: 'private',
      options: ['foo', 'bar'],
    });

    return reviewQuestion.save()
      .then((result) => {
        const {
          _id,
        } = result.toJSON();
        return ReviewQuestion.findById(_id);
      })
      .then((doc) => {
        expect(`${doc._id}`).toBe(`${reviewQuestion._id}`);
        expect(`${doc.i18nId}`).toBe(`${reviewQuestion.i18nId}`);
        expect(`${doc.type}`).toBe(`${reviewQuestion.type}`);
      });
  });
});
