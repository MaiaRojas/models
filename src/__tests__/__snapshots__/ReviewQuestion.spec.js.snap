// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerQuestion should fail when type is unknown 1`] = `
Object {
  "type": [ValidatorError: \`other\` is not a valid enum value for path \`type\`.],
}
`;

exports[`ReviewerQuestion should fail when visibility is unknown 1`] = `
Object {
  "visibility": [ValidatorError: \`other\` is not a valid enum value for path \`visibility\`.],
}
`;

exports[`ReviewerQuestion should fail with missing props 1`] = `
Object {
  "i18nId": [ValidatorError: Path \`i18nId\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
  "visibility": [ValidatorError: Path \`visibility\` is required.],
}
`;
